### DISCUSSION TITLE
Control plane node without taints from machineconfig

### DISCUSSION THREAD

**Discussion by paucampana** [siderolabs/talos#6200] [Q&A] [answered]
Hello,

I am trying to create a single node cluster (just a control-plane, without workers). I would like to be able to run pods there. I have seen that in order to achieve it, I need to:
1. set the machineconfig.controlplane.scheduler.disabled to false in the control plane machineconfig
2. Delete the taints from the node. 

I see in the [documentation ](https://www.talos.dev/v1.2/advanced/troubleshooting-control-plane/#what-is-a-control-plane-node) that `Control plane nodes are tainted by default to prevent workloads from being scheduled to control plane nodes.`

I would like to know if there is any options to set in the machineconfig to create the node without the Taints

Thanks!!

**Comment by frezbo** [comment, marked as answer]
you'd only need to set `.cluster.allowSchedulingOnControlPlanes` to  true. This would bring up the machine with the control plane nodes without taints.

Scheduler should not be disabled.

**Reply by paucampana** [reply]
Working. Thanks!

**Reply by larrywalters** [reply]
@frezbo In my case, I also had trouble with the tainting. I started the cluster and then ran `kubectl taint nodes --all node-role.kubernetes.io/master-` to untaint it, before applying any manifests, and it worked. Is this a bad approach?

**Reply by smira** [reply]
there's nothing bad, but Talos will automatically re-apply the taint on next reboot. by the way taint is now `control-plane`, not `master` (since 1.2.0)

**Reply by larrywalters** [reply]
@smira Thanks for the insight, and good to know that the taint has changed. I'm still on 1.1.1. I tried adding the `allowSchedulingOnControlPlanes` item to the machine configuration applied on cluster start, and removing my `kubectl taint ...` command, but I still get an error and the containers don't start.
In my machine config patch:
```
	{
		"op": "add",
		"path": "/cluster/allowSchedulingOnControlPlanes",
		"value": true
	}
```
Error from `kubectl describe ...`
```
  Warning  FailedScheduling  9s     default-scheduler  0/2 nodes are available: 1 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 1 node(s) had volume node affinity conflict. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.
```
Am I doing something wrong?

**Reply by frezbo** [reply]
`allowSchedulingOnControlPlanes` is only supported on talos 1.2.0 and later

**Reply by smira** [reply]
in 1.1.x it was called `allowSchedulingOnMasters`, you can use config reference on talos.dev for a specific version of Talos to figure out the available settings

**Reply by frezbo** [reply]
the old option is `allowSchedulingOnMasters`

**Reply by larrywalters** [reply]
Ah, thank you @smira @frezbo! Understood. I have adjusted my script to use the proper value depending on the version.
